src/overlays/actors/ovl_En_Zo/z_en_zo.c

```diff
@@ -78,41 +92,257 @@ static DamageTable D_8099F4E8 = {
+void EnZo_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnZo* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099EC50.s")
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099ED4C.s")
+void EnZo_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnZo* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099EE24.s")
+    this->actionFunc(this, globalCtx);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    EnZo_LookAtPlayer(this, globalCtx);
+    EnZo_PlayWalkingSound(this, globalCtx);
+    EnZo_UpdateCollider(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/EnZo_Init.s")
+s32 EnZo_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* actor,
+                          Gfx** gfx) {
+    EnZo* this = (EnZo*)actor;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/EnZo_Destroy.s")
+    if (limbIndex == 15) {
+        Matrix_InsertTranslation(1500.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+        Matrix_InsertXRotation_s(this->headRot.y, MTXMODE_APPLY);
+        Matrix_InsertZRotation_s(-this->headRot.x, MTXMODE_APPLY);
+        Matrix_InsertTranslation(-1500.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+    }
+    if (limbIndex == 8) {
+        Matrix_InsertXRotation_s(-this->upperBodyRot.y, MTXMODE_APPLY);
+        Matrix_InsertZRotation_s(-this->upperBodyRot.x, MTXMODE_APPLY);
+    }
+    if ((limbIndex == 8) || (limbIndex == 9) || (limbIndex == 12)) {
+        rot->y += (s16)(Math_SinS(this->limbRotY[limbIndex]) * 200.0f);
+        rot->z += (s16)(Math_CosS(this->limbRotZ[limbIndex]) * 200.0f);
+    }
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/EnZo_Update.s")
+void EnZo_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* actor, Gfx** gfx) {
+    EnZo* this = (EnZo*)actor;
+    Vec3f sp30 = D_8099F5CC;
+    Vec3f zeroVec = D_8099F5D8;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099EFF4.s")
+    if (D_8099F578[limbIndex] >= 0) {
+        Matrix_MultiplyVector3fByState(&zeroVec, &this->unk_364[D_8099F578[limbIndex]]);
+    }
+    if (limbIndex == 15) {
+        Matrix_MultiplyVector3fByState(&sp30, &actor->focus.pos);
+    }
+    if (limbIndex == 4) {
+        Matrix_MultiplyVector3fByState(&zeroVec, &this->leftFootPos);
+    }
+    if (limbIndex == 7) {
+        Matrix_MultiplyVector3fByState(&zeroVec, &this->rightFootPos);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099F15C.s")
+void EnZo_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnZo* this = THIS;
+    s32 i;
+    u8* alloc = GRAPH_ALLOC(globalCtx->state.gfxCtx, 0x1000);
+    u8* allocHead;
+    Gfx* eyeTextures[3] = { 0x060050A0, 0x060058A0, 0x060060A0 };
```

Comment:
Textures and Gfx are completely different things.
Textures are just arrays of data. We currently use the typedef `TexturePtr` when referring to them

Also, we like to use symbols for those instead of raw pointers.

```suggestion
    TexturePtr eyeTextures[] = { D_060050A0, D_060058A0, D_060060A0 };
```

You'll need to add the corresponding externs and add those to undefined_syms,txt